### ISSUE TITLE
t.w.p.j.xmlstream.toResponse's test for the presence of a stanzaType is faulty.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3467]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#3467|
|Type|defect|
|Created|2008-10-11 19:22:37Z|
|Branch|https://github.com/twisted/twisted/tree/jabber-toResponse-3467|

Instead of looking for `stanzaType` it tests for `type`, which is always there.

<details><summary>Searchable metadata</summary>

```
trac-id__3467 3467
type__defect defect
reporter__ralphm ralphm
priority__normal normal
milestone__ 
branch__branches_jabber_toResponse_3467 branches/jabber-toResponse-3467
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1223752957000000 1223752957000000
changetime__1224066637000000 1224066637000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_1"></a>|@ralphm commented|
|-|-|

(In [24968]) Branching to 'jabber-toResponse-3467'

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm removed owner|
|-|-|

Easy fix. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @ralphm|
|-|-|

Instead of using self.failIf, please use self.assertFalse. You can then merge, thanks.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_4"></a>|@ralphm set status to `closed`|
|-|-|

(In [24975]) Ensure toResponse generates a valid stanza if the stanzaType was not specified.

Author: ralphm.
Reviewer: therve.
Fixes [#3467](https://github.com/twisted/twisted/issues/3467).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
